File: CookUTProject/app/src/main/java/com/example/cookutproject/MainActivity.java
package com.example.cookutproject;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;


public class MainActivity extends AppCompatActivity {

    @Override
    /**
     *Fonction se déclenchant lors de l'arrivée sur l'activity
     */
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    /**
     *Renvoie sur l'activité "saisie"
     * @param view ici saisie
     */
    public void changeActivitySaisie(View view){
        Intent intent = new Intent(this, Saisie.class);
        startActivity(intent);
    }

    /**
     *Renvoie sur l'activité "recap"
     * @param view ici recap
     */
    public void changeActivityRecap(View view){
        Intent intent = new Intent(this, Recap.class);
        startActivity(intent);
    }

    /**
     *Renvoie sur l'activité "event"
     * @param view ici event
     */
    public void changeActivityEvent(View view){
        Intent intent = new Intent(this, Event.class);
        startActivity(intent);
    }
}

File: src/main/java/com/example/testbare/MainActivity.kt
package com.example.testbare
import com.example.testbare.fragment.Ajouter_Depense_Fragment
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.lifecycle.coroutineScope
import com.example.testbare.budget.ClasseTemplate
import com.example.testbare.database.AppDatabase
import com.example.testbare.database.entities.Categorie
import com.example.testbare.fragment.InfoFragment
import com.example.testbare.parametre.Parametre_Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity(){

    private val mDeshboardFragment = Ajouter_Depense_Fragment()
    private val mSettingsFragment = Parametre_Fragment()
    private val mInfoFragment = InfoFragment()
    private val mAnalyseFragment = ClasseTemplate()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val  bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        replaceFragment(mInfoFragment)

        //pour commencer sur ce dashboard
        bottomNavigationView.setOnItemSelectedListener   {
            when(it.itemId){
                R.id.dashboard -> replaceFragment(mDeshboardFragment)
                R.id.settings -> replaceFragment(mSettingsFragment)
                R.id.info->replaceFragment(mInfoFragment)
                R.id.Analyse->replaceFragment(mAnalyseFragment)

                else -> false
            }
        }

        val categorieDao = AppDatabase.getDatabase(this).CategorieDao()

        lifecycle.coroutineScope.launch{
            val cat : Categorie = Categorie("Voiture",400)
            categorieDao.insertCategorie(cat)
            val cat2 : Categorie = Categorie("Alimentation",350)
            categorieDao.insertCategorie(cat2)
            val cat3 : Categorie = Categorie("Charge",520)
            categorieDao.insertCategorie(cat3)
            val cat4 : Categorie = Categorie("Loisirs",150)
            categorieDao.insertCategorie(cat4)
        }

    }


    /***
     * méthode permettant de changer de dashboard
     */

    private fun replaceFragment(fragment: Fragment): Boolean {
        if(fragment!=null){
            val transaction = supportFragmentManager.beginTransaction()
       //     setContentView(R.layout.fragment);
            transaction.replace(R.id.fragment_container, fragment)
            transaction.commit()
            return true

        }
        return false
    }

     fun getContexte() : Context{
        return this.getContexte()
    }

}